File: Noiembrie-2023/Week_no3/Meditație 18.11.2023/Polimorfism_OOP.py
class Shape():
	def __init__(self, heigth, width):
		self.heigth = heigth
		self.width = width
	def aria():
		return 0
	def display():
		return 0

class Rectangle( Shape ):
	def __init__(self, heigth, width):
		self.heigth = heigth
		self.width = width
	def aria(self):
		return self.heigth * self.width 
	def display(self):
		print("The Rectangle is equal %.2lf, %.2lf" %(self.heigth, self.width))

class Triangle( Shape ):	
	def __init__(self, base, width):
		self.base = base
		self.width = width
	def aria(self):
		return self.base * self.width / 2
	def display(self):
		print("The Triangle is equal %.2lf, %.2lf" %(self.base, self.width))

class Circle( Shape ):

	def __init__(self, radius):
		self.radius = radius
	def aria(self):
		return 3.14 * self.radius * self.radius
	def display(self):
		print("The Circle is equal %.2lf " %(self.radius))

def main():
	Ob_r = Rectangle(45, 56)
	Ob_t = Triangle(56, 78)
	Ob_c = Circle(78)

	Ob_r.display()
	print("%.2f\n" %Ob_r.aria())
	Ob_t.display()
	print("%.2f\n" %Ob_t.aria())
	Ob_c.display()
	print("%.2f\n" %Ob_c.aria())

main()

File: Noiembrie-2023/Week_no3/Meditație 18.11.2023/Polimorfism_OOP.cpp
#include<iostream>
using namespace std;
/* OVERRIDE (suprascriere/redefinire)
 OVERLOADING (supraîncărcare de simboluri)
Shape este clasa abstracta, iar aria() si display() sunt functii pure, adica pot fi
redefinite in clasa derivata
*/
class Shape {
public:
	virtual double aria() = 0 ;
	virtual void display() = 0;
};
class Rectangle: public Shape {
private:
	double heigth, width;
public:
	Rectangle(double heigth, double width )
	{
		this->heigth = heigth;
		this->width  = width;
	}
	double aria(){
		return heigth * width;
	}
	void display()
	{
		printf("Rectangle is equal: heigth= %.2lf, width= %.2lf ", heigth, width );
	}
};
class Triangle: public Shape {
private:
	float base, heigth;
public:
	Triangle(float base, float heigth) {
		this-> base = base;
		this->heigth= heigth;
	}
	double aria() {
		return base*heigth/2;
	}
	void display() {
		printf("Triangle is equal: base = %.2lf, heigth = %.2lf ", base, heigth);
	}
};
class Circle: public Shape {
private:
	float radius;
public:
	Circle(float r) {
		radius = r;
	}
	double aria() {
		return 3.14 * radius * radius;
	}
	void display() {
		printf("Circle is equal: %.2lf ", radius);
	}
};
int main()
{
	Shape *r = new Rectangle(34, 45);
	Shape *t = new Triangle(6,7);
	Shape *c = new Circle(4);
	
	r->display();
	printf("%.2lf\n", r->aria());
	
	t->display();
	printf("%.2lf\n", t->aria());
	
	c->display();
	printf("%.2lf\n", c->aria());
}



/*	În C avem: lf(long float) pentru double, f pt float
               d = decimal, c = character, s=string
	În python avem:  f pt double(sau float), c = character, s= string, d = decimal  
În C:
int main()
{
    double a = 1.23;
    float b = 3.14;
    printf( "%.2lf %.2lf", a, b );
    
    return 0;
}
În py:
def main():
    
    a = 1.23
    b = 3.14
    c ="Andrei"
    d= "Boto"
    print( "%.2lf %.2lf" %(a, b) )
    print("%s %s" %(c, d) )
main()
*/

